### ISSUE TITLE
Unparsable section with Jinja code in dbt models with sqlfluff lint functionality

### ISSUE THREAD

**Issue by ltsyen** [sqlfluff/sqlfluff#7223]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running sqlfluff lint, we're encountering issues that some sections are deemed 'unparsable':

== [src\models\staging\adventure\xxxxxxx.sql] FAIL
L:   3 | P:  14 |  PRS | Line 3, Position 14: Found unparsable section: "from {{
                       | ref('xxxxxx') }}"

### Expected Behaviour

* I would expect that the jinja and dbt specific code pieces would not cause any issues

### Observed Behaviour

* Cannot parse the dbt code with jinja {{ref()}} function to refer to other models.

I was running on windows 11.

I tried running on WSL with completely the same setup and there everything runs successfully. So I assume it has something to do with the windows distribution.

### How to reproduce

```bash
sqlfluff lint src/models/marts/common/dim_date.sql

=== [dbt templater] Sorting Nodes...
WARNING:thrift.transport.sslcompat:using legacy validation callback
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [src\models\marts\common\dim_date.sql] FAIL
L:   3 | P:  14 |  PRS | Line 3, Position 14: Found unparsable section: "from {{
                       | ref('stg_adventure__date') }}"
WARNING: Parsing errors found and dialect is set to 'databricks'. Have you configured your dialect correctly?
```

**sql file example:**

```sql
with stg_date as (

    select * from {{ ref('stg_adventure__date') }}

)

select * from stg_date
```

### Dialect

databricks

### Version

* Python 3.13.1
* sqlfluff, version 3.5.0
* pre-commit 4.3.0
* dbt --version 
Core:
  - installed: 1.10.13
  - latest:    1.10.13 - Up to date!

Plugins:
  - databricks: 1.10.14 - Up to date!
  - spark:      1.9.3   - Up to date!

### Configuration

**.sqlfluff file:**
```
[sqlfluff]
dialect = databricks
templater = dbt
runaway_limit = 5
max_line_length = 110
indent_unit = space

[sqlfluff:templater:dbt]
project_dir = .
```

**.pre-commit-config.yaml file:**
```- repo: https://github.com/sqlfluff/sqlfluff
  rev: 3.5.0
  hooks:
    - id: sqlfluff-lint
      # For dbt projects, this installs the dbt "extras".
      # You will need to select the relevant dbt adapter for your dialect
      # (https://docs.getdbt.com/docs/available-adapters):
      # additional_dependencies: ['<dbt-adapter>']
      additional_dependencies : ['dbt-databricks', 'sqlfluff-templater-dbt']
    - id: sqlfluff-fix
      # Arbitrary arguments to show an example
      # args: [--rules, "LT02,CP02"]
      additional_dependencies: ['dbt-databricks', 'sqlfluff-templater-dbt']
```

**requirements-dev.txt**
```
dbt-databricks>=1.8.0,<2.0.0
pre-commit
sqlfluff
sqlfluff-templater-dbt
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by MRaza98** [comment]
I have the same issue!

**Comment by MRaza98** [comment]
which version of sqlfluff-templater-dbt do you have?

**Comment by ltsyen** [comment]
> which version of sqlfluff-templater-dbt do you have?

sqlfluff_templater_dbt-3.5.0

**Comment by alisdairjsmyth** [comment]
I have the same issue. dbt compilation is successful, but not parsing builtin dbt macros.

```
Windows 11
Python 3.12.3
dbt-core                  1.10.19
dbt-databricks            1.10.19
sqlfluff                  4.0.0
sqlfluff-templater-dbt    4.0.0
sqlfluffrs                4.0.0
```

Config in `pyproject.toml`
```
[tool.sqlfluff.core]
templater = "dbt"
dialect = "databricks"
max_line_length = 88

[tool.sqlfluff.templater.dbt]
project_dir = "."
dbt_skip_compilation_error = false

[tool.sqlfluff.templater.jinja]
apply_dbt_builtins = true

```

**Comment by ltsyen** [comment]
This was actually silently fixed in 4.1.0. 
Yet, it contains no test cases that would capture this.
